### PR TITLE
M2 2848

### PR DESCRIPTION
**Pull request by ksuh90**

### CODE REVIEW

File: jerry/util/jira_helper.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class JiraHelper:
+    """
+    Implements a wrapper over Jira which defines higher level jira actions.
+
+    Attributes:
+        jira: A JIRA instance
+    """
+    
+    def __init__(self, site, user, pw):
+        self.jira = JIRA(options={'server': site}, basic_auth=(user, pw))
```
Comment (calebgrana) [inline jerry/util/jira_helper.py]: as mentioned, should probably use oauth in the future

File: jerry/commands/close_released_issues.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    def run(self):
+        super(CloseReleasedIssues, self).run()
+
+        jira_helper = JiraHelper(JIRA_CONFIG['site'],
+                                JIRA_CONFIG['user'],
+                                JIRA_CONFIG['password'])
+
+        issues = jira_helper.get_issues_by_fix_version(self.args.version)
+
```
Comment (calebgrana) [inline jerry/commands/close_released_issues.py]: like the flexibility to only release the tickets by fixVersion. but better if instead the workflow was:
- get all "ready to release" tickets
- assign a fixVersion to them
- close them

File: jerry/commands/send_release_email.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                                JIRA_CONFIG['user'],
+                                JIRA_CONFIG['password'])
+        issues = jira_helper.get_issues_by_fix_version(self.args.version)
+        issue_map = self.get_issues_by_type(issues)
+        email_template = 'release-email.html'
+        email_html = self.get_body_html(email_template, {
+            'fix_version': self.args.version,
+            'issue_map': issue_map.items(),
+            'site_url': JIRA_CONFIG['site'],
+            'downtime': self.args.downtime,
```
Comment (calebgrana) [inline jerry/commands/send_release_email.py]: in the future, we may want to specify the actual `downtime` period as the input?

File: jerry/util/jira_helper.py (hunk 1 above)
Comment (ksuh90) [reply jerry/util/jira_helper.py]: noted.

File: jerry/commands/send_release_email.py (hunk 3 above)
Comment (ksuh90) [reply jerry/commands/send_release_email.py]: fixed. changed argument to accept value.

File: jerry/commands/close_released_issues.py (hunk 2 above)
Comment (ksuh90) [reply jerry/commands/close_released_issues.py]: fixed. added new command "assign-fix-version".

File: jerry/commands/close_released_issues.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+from jerry.config import JIRA_CONFIG
+
+
+class CloseReleasedIssues(Base):
+    """
+    Implements 'jerry close-released-issues'.
+    """
+
+    # The status id for "Close"
+    STATUS_ID = '191'
```
[the remark this answers is not in the document]
Comment (ksuh90) [reply jerry/commands/close_released_issues.py]: fixed

File: jerry/commands/send_release_email.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        Create a map, with the given array of issues, in the form of:
+            {
+                type1: [<issue>, <issue>, ...],
+                type2: [...],
+                ...
+            }
+        """
+        issue_map = {}
+        for issue in jira_issues:
+            issue_map.setdefault(issue.fields.issuetype.name, []).append(issue)
```
[the remark this answers is not in the document]
Comment (ksuh90) [reply jerry/commands/send_release_email.py]: fixed.

File: jerry/commands/close_released_issues.py (hunk 2 above)
Comment (ksuh90) [reply jerry/commands/close_released_issues.py]: Managed to handle the suggested flow. Perhaps in this order:
jerry assign-fix-version
jerry close-issues
jerry git tag
jerry send-release-email

**Pull request merged**